packages/components/src/internal/components/forms/input/SampleStatusInput.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { FC, memo, ReactNode, ReactText, useCallback, useEffect, useState, useMemo } from 'react';
+
+import {
+    DISCARD_CONSUMED_CHECKBOX_FIELD,
+    DISCARD_CONSUMED_COMMENT_FIELD,
+    DiscardConsumedSamplesPanel,
+} from '../../samples/DiscardConsumedSamplesPanel';
+
+import { Alert } from '../../base/Alert';
+import { QueryColumn } from '../../../../public/QueryColumn';
+import { QuerySelect } from '../QuerySelect';
+import { SampleStateType } from '../../samples/constants';
+import { ComponentsAPIWrapper, getDefaultAPIWrapper } from '../../../APIWrapper';
+
+interface SampleStatusInputProps {
+    api?: ComponentsAPIWrapper;
+    col: QueryColumn;
+    data: any;
+    key: ReactText;
+    isDetailInput?: boolean;
+    allowDisable?: boolean;
+    initiallyDisabled?: boolean;
+    onToggleDisable?: (disabled: boolean) => void;
+    value?: string | Array<Record<string, any>>;
+    onQSChange?: (name: string, value: string | any[], items: any) => void;
+    renderLabelField?: (col: QueryColumn) => ReactNode;
+    showAsteriskSymbol?: boolean;
+    onAdditionalFormDataChange?: (name: string, value: any) => any;
+    inputClass?: string;
+    formsy?: boolean; // for jest test
+    forceShowDiscard?: boolean; // workaround for jest test due to difficulty getting QuerySelect to work with jest
+}
+
+export const SampleStatusInput: FC<SampleStatusInputProps> = memo(props => {
+    const {
+        api,
+        showAsteriskSymbol,
+        allowDisable,
+        col,
+        key,
+        initiallyDisabled,
+        onToggleDisable,
+        value,
+        onQSChange,
+        renderLabelField,
+        onAdditionalFormDataChange,
+        inputClass,
+        formsy,
+    } = props;
+    const [consumedStatuses, setConsumedStatuses] = useState<number[]>(undefined);
+    const [error, setError] = useState<string>(undefined);
+    const [showDiscardPanel, setShowDiscardPanel] = useState<boolean>(false);
+    const [shouldDiscard, setShouldDiscard] = useState<boolean>(true);
+
+    const loadConsumedStatuses = async (): Promise<void> => {
+        try {
+            const statuses = await api.samples.getSampleStatuses();
+            const consumedStatusIds = [];
+            statuses.forEach(status => {
+                if (status.stateType == SampleStateType.Consumed) consumedStatusIds.push(status.rowId);
+            });
+            setConsumedStatuses(consumedStatusIds);
+        } catch (error) {
+            console.error(error.exception);
+            setError('Error loading sample statuses');
```

Comment:
Good idea to give more details. Since the error case is not expected to happen if things work correctly, we probably don't need to worry about showing too much text on error.